cpp/src/arrow/ipc/json_simple.cc

```diff
@@ -410,6 +410,41 @@ class DayTimeIntervalConverter final
   std::shared_ptr<DayTimeIntervalBuilder> builder_;
 };
 
+class MonthDayNanoIntervalConverter final
+    : public ConcreteConverter<MonthDayNanoIntervalConverter> {
+ public:
+  explicit MonthDayNanoIntervalConverter(const std::shared_ptr<DataType>& type) {
+    this->type_ = type;
+    builder_ = std::make_shared<MonthDayNanoIntervalBuilder>(default_memory_pool());
+  }
+
+  Status AppendValue(const rj::Value& json_obj) override {
+    if (json_obj.IsNull()) {
+      return this->AppendNull();
+    }
+    MonthDayNanoIntervalType::MonthDayNanos value;
+    if (!json_obj.IsArray()) {
+      return JSONTypeError("array", json_obj.GetType());
+    }
+    if (json_obj.Size() != 3) {
+      return Status::Invalid(
+          "month day nanos interval pair must have exactly two elements, had ",
```

Comment:
Hmm, I think the error message is slightly off.
Also, I think we need to settle on a spelling of the type name. Apparently it's "month_day_nano_interval". Let's not vary gratuitously :-)

Reply:
sorry, fixed.